### ISSUE TITLE
Can't parse Json to JSONModel using initWithString

### ISSUE THREAD

**Issue by issamla** [icanzilb/JSONModel#309]
I can't parse a Json using initWithString, i get this Error : 

    [__NSCFDictionary dataUsingEncoding:]: unrecognized selector sent to instance

My request : 

    [self.httpClient POST:URL parameters:params success:^(AFHTTPRequestOperation *operation, id responseObject) {

    Member *member = [[Member alloc] initWithString:responseObject[@"member"] error:nil];

And member returns nil.

Json Response : 

    {
    access_token: "KNJND8ZbHNUFdhWdfdgdgH5DVUQ1kAu3FwAYms0AN"
    token_type: "bearer"
    expires: 1429008129
    expires_in: 604800
    refresh_token: "xevNQhZJeS5QddgfhghgqRPTgF0B23CGNrYQ3IwdW"
    member: {
        id: 785
        email: "issam@geeksters.co"
        full_name: "aaaaaa"
        address: "settatjhvbjkhbkn"
        phone: "0651536649"

Member Model : 

    @interface Member : JSONModel
      @property (strong, nonatomic) NSNumber<Optional> *id;
      @property (strong, nonatomic) NSString<Optional> *email;
      @property (strong, nonatomic) NSString *full_name;
      @property (strong, nonatomic) NSString<Optional> *address;
      @property (strong, nonatomic) NSString<Optional> *phone;

**Comment by gitabdelkarim** [comment]
Try " initWithDictionary" : 
     Member *member = [[Member alloc] initWithDictionary:responseObject[@"member"] error:nil];

**Comment by issamla** [comment]
i tried this : 

    Member *member = [[Member alloc] initWithDictionary:responseObject[@"member"] error:nil];

but the member still null.

**Comment by sauvikatinnofied** [comment]
Let me know the JSONParsing error if there any, use the following code

```objective-c
NSError *jsonParsingError;
Member *member = [[Member alloc] initWithDictionary:responseObject[@"member"] error:&jsonParsingError];
if(jsonParsingError){
  NSLog(@"JSONParsing Error: %@", error.localizedDescription);
}else{
 NSLog(@"Member created successfully");
}

**Comment by sauvikatinnofied** [comment]
If possible give us the JSON response full so that we can create the Member object from locally stored json file.

**Comment by jasonmarziani** [comment]
Think I figured out this issue.  I had a child model that had params in Objective-C missing from the JSON and not marked as optional.  I was getting this Exception when trying to access any params on the child objects, but no parse errors.  Once I tried to create the child objects directly, I got the parse error about the missing params, fixed the issues, and now its no longer throwing Exceptions.
